File: Nth Fibonacci Number.cpp
  const int M = 1e9+7;
    int nthFibonacci(int n){
        // code here
         long long n1 = 0;
      long long n2 = 1;
      
      for(int i=2;i<=n;i++){
          long long n3 = n2;
          n2 = (n1+n2)%M;
          n1 = n3;
      }
      return n2;
        
    }

File: Nth Fibonacci Number.java


//User function Template for Java

//User function Template for Java
class Solution {
    static int nthFibonacci(int n){
        // code here
        int a=0;
        int b=1;
        if(n==0){
            return a;
        }
        int c;
        for(int i=2;i<=n;i++){
            c=(a+b)%1000000007;
            a=b;
            b=c;
        }
        
        return b;
    
    }
}
